### PR TITLE
Release candidate for ACCP 2.3.0

### PR DESCRIPTION
**Pull request by amirhosv**
*Description of changes:*

Release candidate for ACCP 2.3.0

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/gradle-kt-dsl/lib/build.gradle.kts [hunk 1]
```diff
@@ -28,10 +29,18 @@ dependencies {
     // Use the Kotlin JUnit integration.
     testImplementation("org.jetbrains.kotlin:kotlin-test-junit")
 
+    val accpArtifactId =
+        if (project.hasProperty("fips"))
+            "AmazonCorrettoCryptoProvider-FIPS"
+        else
+            "AmazonCorrettoCryptoProvider"
+
+    println(accpArtifactId)
```
Comment (geedo0) [inline examples/gradle-kt-dsl/lib/build.gradle.kts:38]: Same here.

File: README.md [hunk 2]
```diff
@@ -79,6 +89,21 @@ KeyFactory algorithms:
 * RSA
 * ACCP's SecureRandom uses AWS-LC's DRBG implementation.
 
+# Notes on ACCP-FIPS
+ACCP-FIPS is a variation of ACCP which is uses AWS-LC built in FIPS-mode.
```
Comment (brian-jarvis-aws) [inline README.md:93]: "which is uses" --> "which uses"
I acknowledge that I caused this typo as it was in my original suggested phrasing too.

File: build.gradle [hunk 3]
```diff
@@ -24,6 +24,26 @@ if (ext.lcovIgnore == null) {
...
+    return [
+        'The Amazon Corretto Crypto Provider FIPS (ACCP-FIPS) is a collection',
+        'of high-performance cryptographic implementations exposed via the',
+        'standard JCA/JCE interfaces. ACCP-FIPS uses AWS-LC built in FIPS mode',
+        'and is provided for experimental purposes only.'
+    ].join(" ")
+
+}
+
+println getDescriptionForPom(ext.isFips)
```
Comment (geedo0) [inline build.gradle:45]: Was this leftover from development? Suggest dropping it as we don't really log this kind of stuff anywhere else.

File: README.md [hunk 4]
```diff
@@ -79,6 +89,21 @@ KeyFactory algorithms:
...
+AWS-LC is undergoing FIPS validation testing by an accredited lab and,
+upon completion, will be submitted to NIST for certification.
+This process can take a long time, so we provide ACCP-FIPS for
+experimentation purposes in the interim.
+
+Version 2.3.0 is the first release of ACCP-FIPS. The Maven coordinates for
+ACCP-FIPS is the same as ACCP with one difference that ACCP-FIPS's
```
Comment (brian-jarvis-aws) [inline README.md:100]: "is the same" --> "are the same" (plural agreement with "coordinates")

File: build.gradle [hunk 5]
```diff
@@ -660,6 +680,7 @@ task fakePublish {
 
 }
 
+
```
Comment (geedo0) [inline build.gradle:683]: nit: extra newline

File: examples/gradle-kt-dsl/run-tests.sh [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-NUMBER_OF_RETRIES=180
+# Retry the build for 5 hours, every 10 seconds
+NUMBER_OF_RETRIES=1800
 DELAY_BETWEEN_RETRIES=10
 
 for i in $(seq 1 ${NUMBER_OF_RETRIES})
 do
     echo "Iteration ${i}"
-    ./gradlew lib:test
+    ./gradlew lib:test && ./gradlew -Pfips lib:test
```
Comment (geedo0) [inline examples/gradle-kt-dsl/run-tests.sh:10]: Would be nice if this script could tell you whether the standard or fips build broke.

File: README.md [hunk 7]
```diff
@@ -64,13 +65,22 @@ Signature algorithms:
...
 * ECDH
 
+SecretKeyFactory:
+* HkdfWithHmacSHA1
+* HkdfWithHmacSHA256
+* HkdfWithHmacSHA384
+* HkdfWithHmacSHA512
+
 SecureRandom algorithms:
 * ACCP's SecureRandom uses AWS-LC's DRBG implementation, which is described [here](https://github.com/awslabs/aws-lc/blob/main/third_party/jitterentropy/README.md) and [here](https://github.com/awslabs/aws-lc/blob/725625435158150ef21e0a4dab6fa3aca1ef2d2c/crypto/fipsmodule/rand/rand.c#L36-L60).
```
Comment (geedo0) [inline README.md:85]: Is it worth calling out that AWS-LC's secure random changes significantly based on the FIPS flag or should we trust our readers to click through and read the comments here? I forget if we called that out in detail somewhere else.

**Review by geedo0 — APPROVED** [review]
Approved with minor comments.

File: build.gradle (hunk 3 above)
Comment (amirhosv) [reply build.gradle:45]: You got me. I'll remove those.

File: examples/gradle-kt-dsl/lib/build.gradle.kts (hunk 1 above)
Comment (amirhosv) [reply examples/gradle-kt-dsl/lib/build.gradle.kts:38]: done

File: build.gradle (hunk 3 above)
Comment (amirhosv) [reply build.gradle:45]: done

File: build.gradle (hunk 5 above)
Comment (amirhosv) [reply build.gradle:683]: done

File: README.md (hunk 2 above)
Comment (amirhosv) [reply README.md:93]: done

File: README.md (hunk 4 above)
Comment (amirhosv) [reply README.md:100]: done

File: examples/gradle-kt-dsl/run-tests.sh (hunk 6 above)
Comment (amirhosv) [reply examples/gradle-kt-dsl/run-tests.sh:10]: From the log is clear which one fails. Having said that, this script is to check that both artifacts are available on Maven.

File: README.md (hunk 7 above)
Comment (amirhosv) [reply README.md:85]: I added a note.

Comment (brian-jarvis-aws) [reply README.md:85]: There is more detail on the differences in SecureRandom behavior on the [associated property](https://github.com/corretto/amazon-corretto-crypto-provider#other-system-properties)'s description.

**Review by WillChilds-Klein — DISMISSED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## 2.3.0
+
+### Overview
+
+Starting from this version, build artifacts for ACCP-FIPS are also released for
```
Comment (darylmartin100) [inline CHANGELOG.md:7]: Can you document the branch that you're using along with the commit?

File: README.md [hunk 9]
```diff
@@ -64,20 +65,47 @@ Signature algorithms:
...
+
+AlgorithmParameters:
+* EC. Please refer to [system properties](https://github.com/corretto/amazon-corretto-crypto-provider#other-system-properties) for more information.
+
+
+# Notes on ACCP-FIPS
+ACCP-FIPS is a variation of ACCP which uses AWS-LC built in FIPS-mode.
+AWS-LC is undergoing FIPS validation testing by an accredited lab and,
+upon completion, will be submitted to NIST for certification.
+This process can take a long time, so we provide ACCP-FIPS for
```
Comment (darylmartin100) [inline README.md:99]: I would rephrase this sentence to highlight that we are releasing ACCP-FIPS to allow users to do early performance testing before we are granted our certificate.  I would also add a note in here to call out where future updates for the status of ACCP-FIPS will be.  Blog post?

File: README.md [hunk 10]
```diff
@@ -64,20 +65,47 @@ Signature algorithms:
...
+experimentation purposes in the interim.
+
+Version 2.3.0 is the first release of ACCP-FIPS. The Maven coordinates for
+ACCP-FIPS are the same as ACCP with one difference that ACCP-FIPS's
+artifact ID is `AmazonCorrettoCryptoProvider-FIPS`.
+
+Notable differences between ACCP and ACCP-FIPS:
```
Comment (darylmartin100) [inline README.md:106]: You should call out the performance difference if they choose to use the Jitter backed entropy source and also other small performance differences.

Comment (brian-jarvis-aws) [reply README.md:106]: Those performance impacts are already discussed on the description of the [system property](https://github.com/corretto/amazon-corretto-crypto-provider#other-system-properties) that this links to. I think we should keep it in a single place, since it is prone to becoming stale or out of sync the more places it's documented.

**Review by darylmartin100 — APPROVED** [review]

**Review by WillChilds-Klein — APPROVED** [review]
